packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1468,6 +1478,38 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
     return SelectionUtils.getResultBasedOnRect(_rect, localPosition);
   }
 
+  SelectionResult _updateSelectionEdgeByWord(Offset globalPosition, {required bool isEnd}) {
+    _setSelectionPosition(null, isEnd: isEnd);
+    final Matrix4 transform = paragraph.getTransformTo(null);
+    transform.invert();
+    final Offset localPosition = MatrixUtils.transformPoint(transform, globalPosition);
+    if (_rect.isEmpty) {
+      return SelectionUtils.getResultBasedOnRect(_rect, localPosition);
+    }
+    final Offset adjustedOffset = SelectionUtils.adjustDragOffset(
+      _rect,
+      localPosition,
+      direction: paragraph.textDirection,
+    );
+    final TextPosition position = _clampTextPosition(paragraph.getPositionForOffset(adjustedOffset));
```

Comment:
This probably should instead clamp after _getWordBoundaryAtPosition

it is fine if the `position` is outside of the text range of this selectablefragment, as long as the `targetPosition` is in the text range.